src/ui/component/copyToClipboard/copyToClipboard.tsx

```diff
@@ -1,32 +1,54 @@
+import { useCallback, useState } from 'react'
 import { useAppStore } from '@/ui/store/appStore'
-import { useCallback } from 'react'
+import { useDebounce } from '@/ui/hook/useDebounce'
 import { Icon } from '@/ui/component/icon/icon'
+import classNames from 'classnames'
 import './copyToClipboard.scss'
 
+type CopyState = 'copy' | 'success' | 'error'
+
 type CopyToClipBoardProps = {
   textToCopy: string
   onCopied?: (isCopied: boolean) => void
 }
 
 export const CopyToClipboard = ({ onCopied, textToCopy }: CopyToClipBoardProps): JSX.Element => {
   const theme = useAppStore(state => state.theme)
+  const [copyState, setCopyState] = useState<CopyState>('copy')
+
+  const resetCopyState = useCallback((): void => {
+    setCopyState('copy')
+  }, [])
+
+  useDebounce(resetCopyState, copyState !== 'copy' ? 2000 : null, [copyState])
```

Comment:
As an improvement, make the constant `2000` a parameter of the component with a default value.